Automattic-Tracks-iOS/Services/EventLoggingUploadQueue.swift

```diff
@@ -0,0 +1,43 @@
+import Foundation
+
+open class EventLoggingUploadQueue {
+
+    private let fileManager = FileManager.default
+
+    /// The log file on top of the queue
+    var first: LogFile? {
+
+        guard let url = try? fileManager.contentsOfDirectory(at: storageDirectory).first else {
+            return nil
+        }
+
+        return LogFile(url: url, uuid: url.lastPathComponent)
+    }
+
+    func add(_ log: LogFile) throws {
+        try ensureStorageDirectoryExists()
+        try fileManager.copyItem(at: log.url, to: storageURL(forLog: log))
+    }
+
+    func remove(_ log: LogFile) throws {
+        let url = storageURL(forLog: log)
+        if fileManager.fileExistsAtURL(url) {
+            try fileManager.removeItem(at: storageURL(forLog: log))
+        }
+    }
+
+    var storageDirectory: URL {
+        return fileManager.documentsDirectory.appendingPathComponent("log-upload-queue")
+    }
+
+    func storageURL(forLog log: LogFile) -> URL {
+        let newURL = storageDirectory.appendingPathComponent(log.uuid)
+        return newURL
+    }
+
+    func ensureStorageDirectoryExists() throws {
```

Comment:
Nitpick: Before checking the implementation of this method I thought that it was going to crash if the directory didn't exist. How do you feel about renaming it to something like `createStorageDirectoryIfItDoesntExist` instead?

Reply:
Works for me – changed in 92afe2d422968422c12f367bfe0d1b09cce1c3e9.